### ISSUE TITLE
nftables + banaction 'allports' creates non-functional matching rule

### ISSUE THREAD

**Issue by zenlord** [fail2ban/fail2ban#2782]
### Environment:
- Debian 4.19.118-2+deb10u1 (2020-06-07) x86_64 GNU/Linux
- fail2ban 0.10.2
- nftables v0.9.0 (Fearless Fosdick)

- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

I have set up a few jails, and noticed that my asterisk jail was filled with banned IP's, but none of them were actually banned. 

# nft list tables
```
table inet fail2ban
table inet filter
```

This is how the 'input chain' looked like when using the nftables_allports banaction:
```
        chain input {
                type filter hook input priority 100; policy accept;
                meta l4proto tcp meta nfproto ipv4 ip saddr @f2b-asterisk-udp drop
                meta l4proto tcp meta nfproto ipv4 ip saddr @f2b-asterisk-tcp drop
                tcp dport { imaps } ip saddr @f2b-dovecot drop
        }
```
-> There seem to be 2 match rules inserted/mixed ('meta l4proto' and 'meta nfproto').

Switching to the default banaction for my asterisk jail resulted in this working configuration:

# nft list table inet fail2ban
```
table inet fail2ban {
	set f2b-asterisk-tcp {
		type ipv4_addr
		elements = { <IPs> }
	}

	set f2b-asterisk-udp {
		type ipv4_addr
		elements = { <IPs> }
	}

	chain input {
		type filter hook input priority 100; policy accept;
		udp dport { sip, sip-tls } ip saddr @f2b-asterisk-udp drop
		tcp dport { sip, sip-tls } ip saddr @f2b-asterisk-tcp drop
	}
}
```

My default action.d/nftables-allports.conf:
```
[INCLUDES]
before = nftables-common.conf
[Definition]
nftables_mode = meta l4proto <protocol>
[Init]
```

And my customized action.d/nftables-common.local:
```
[Init]
nftables_family = inet
nftables_table = fail2ban
blocktype = drop
```

I have done a fair amount of searching, but did not find anything relevant to help me conclude whether this was a bug in v0.10.2, or whether this behaviour is the result of a misconfiguration on my behalf. Happy to forward this to the Debian folks if this is a (known) bug.

**Comment by sebres** [comment]
I can't follow what exactly is the issue? Do you see some errors in fail2ban.log? What exactly "non-functional matching rule" means?

Also note that nftables actions get completely rewritten in #2254. So you can try new version of action-config (and some new parameters).

Also note similar issue for configuration example - #2760.

**Comment by zenlord** [comment]
Thank you sebres, for your comment.

As I wrote above: I'm not entirey sure whether this behavior was caused by a bug in the Debian Buster version of fail2ban, or a configuration error. The result was that my asterisk jail did insert offending ip's into a dedicated nft table, but those ip's could just happily continue connecting to my asterisk server. In other words: the banaction 'allports' was not functioning correctly.

The observation that triggered my issue report here is that the line

    meta l4proto tcp meta nfproto ipv4 ip saddr @f2b-asterisk-udp drop

seems to combine/mix two nftables matching rules:

    meta l4proto tcp ip saddr @f2b-asterisk-udp drop
    meta nfproto ipv4 ip saddr @f2b-asterisk-udp drop

I'm sure (if this was a bug at some point in time), it has been fixed, reading those links that you referenced here.
Thank you!
